e2e/Utils.ts

```diff
@@ -260,3 +261,12 @@ export const normalizeIcons = (str: string) => {
     .replace(new RegExp('\u00D7', 'g'), '\u2715')
     .replace(new RegExp('\u221A', 'g'), '\u2713');
 };
+
+// Certain environments (like CITGM and GH Actions) do not come with mercurial installed
+export const buildTestIfHg = () => {
+  const hgIsInstalled = which.sync('hg', {nothrow: true}) !== null;
+  if (!hgIsInstalled) {
+    console.warn('Mercurial (hg) is not installed - skipping some tests');
+  }
+  return hgIsInstalled ? test : test.skip;
+};
```

Comment:
```suggestion
let hgIsInstalled: boolean | null = null;

export const testIfHg: typeof test = (...args) => {
  if (hgIsInstalled === null) {
    hgIsInstalled = which.sync('hg', {nothrow: true}) !== null;
  }
  
  if (hgIsInstalled) {
    test(...args);
  } else {
    console.warn('Mercurial (hg) is not installed - skipping some tests');

    test.skip(...args);
  }
};
```

Reply:
Actually the type of `testIfHg` should not be `typeof test` as `jest.It` is more than just a function:

```ts
interface It {
    (name: string, fn?: ProvidesCallback, timeout?: number): void;
    only: It;
    skip: It;
    todo: It;
    concurrent: It;
    each: Each;
}
```

Our type is:

```ts
(...args: Parameters<typeof test>) => ReturnType<typeof test>
```

I just fixed it on my last commit.